src/types/mod.rs

```diff
@@ -121,6 +121,28 @@ impl System {
     pub const fn get_all() -> [System; 4] {
         [System::CH, System::AT, System::IT, System::SLO ]
     }
+
+    // returns neighboring countries for a given system
+    pub fn adjacent(&self) -> Vec<System> {
+        match self {
+            System::CH => vec![System::AT, System::IT],
+            System::AT => vec![System::CH, System::IT, System::SLO],
+            System::IT => vec![System::AT, System::CH, System::SLO],
+            System::SLO => vec![System::AT, System::IT],
+        }
+    }
+    // returns an array of matching neighboring countries for two systems
+    pub fn shared_adjacency(&self, dest: System) -> Vec<System> {
```

Comment:
I'm not 100% happy with how this is implemented. It's correct in that it returns systems that are adjacent by proxy, however it is not really practical to use.

We have two separate calls for direct and indirect adjacency, which return results from different types. This makes it very complicated when we actually need this information. E.g. if we call from our path-finding algorithm we call two different functions that return two different types.

I added a more idiomatic solution to the problems that returns all the information wrapped inside an Enum that tells us whether an adjacency is direct or indirect.